### PR TITLE
Upgraded Sentry Version and added support for User Feedback

### PR DESCRIPTION
**Pull request by backstageel**
I, 

I updated Sentry to the latest version and added support for user feedback using the Sentry Modal.

All you have to do is add the following code to your error500 and error400 page:

`<script src="https://cdn.ravenjs.com/2.3.0/raven.min.js"></script>
<script>
    Raven.showReportDialog({
        eventId: '<?php echo CakeSession::read('sentry_event_id')?>',

        // use the public DSN (dont include your secret!)
        dsn: '<?php echo Configure::read('Sentry.javascript.server')?>'
    });
</script>`

### CODE REVIEW

**Comment by Sandreu** [comment]
Seems great thank you.

**Comment by mfn** [comment]
@Sandreu would you mean creating a new release?

**Comment by mfn** [comment]
When I tried this PR on a project with console commands, it bombed => I've created https://github.com/Sandreu/cake-sentry/pull/18 with a fix.
```
PHP Warning:  Illegal string offset 'exception' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671

Warning: Illegal string offset 'exception' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671
PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671
PHP Warning:  Illegal string offset 'logger' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 675

Warning: Illegal string offset 'logger' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 675
PHP Warning:  Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 680

Warning: Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 680
PHP Warning:  Illegal string offset 'timestamp' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 849

Warning: Illegal string offset 'timestamp' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 849
PHP Warning:  Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 852

Warning: Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 852
PHP Warning:  Illegal string offset 'tags' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855

Warning: Illegal string offset 'tags' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855
PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855
PHP Warning:  Illegal string offset 'extra' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858

Warning: Illegal string offset 'extra' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858
PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858
PHP Warning:  Illegal string offset 'event_id' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 861

Warning: Illegal string offset 'event_id' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 861
PHP Warning:  array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 868

Warning: array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 868
PHP Warning:  array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 874

Warning: array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 874
PHP Warning:  array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 886

Warning: array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 886
PHP Notice:  Undefined index: extra in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891

Notice: Undefined index: extra in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891
PHP Warning:  array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891

Warning: array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891
PHP Notice:  Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 946

Notice: Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 946
PHP Notice:  Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 948

Notice: Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 948
Warning Error: Illegal string offset 'exception' in [/vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php, line 671]

PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 671
PHP Warning:  Illegal string offset 'logger' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 675

Warning: Illegal string offset 'logger' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 675
PHP Warning:  Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 682

Warning: Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 682
PHP Warning:  Illegal string offset 'timestamp' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 849

Warning: Illegal string offset 'timestamp' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 849
PHP Warning:  Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 852

Warning: Illegal string offset 'level' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 852
PHP Warning:  Illegal string offset 'tags' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855

Warning: Illegal string offset 'tags' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855
PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 855
PHP Warning:  Illegal string offset 'extra' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858

Warning: Illegal string offset 'extra' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858
PHP Notice:  Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858

Notice: Array to string conversion in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 858
PHP Warning:  Illegal string offset 'event_id' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 861

Warning: Illegal string offset 'event_id' in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 861
PHP Warning:  array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 868

Warning: array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 868
PHP Warning:  array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 874

Warning: array_merge(): Argument #2 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 874
PHP Warning:  array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 886

Warning: array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 886
PHP Notice:  Undefined index: extra in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891

Notice: Undefined index: extra in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891
PHP Warning:  array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891

Warning: array_merge(): Argument #3 is not an array in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 891
PHP Notice:  Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 946

Notice: Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 946
PHP Notice:  Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 948

Notice: Undefined index: event_id in /vagrant/project/app/Vendor/sentry/sentry/lib/Raven/Client.php on line 948
Error: this is just another test, all good!
#0 /vagrant/project/lib/Cake/Console/Shell.php(459): MigrationShell->test()
#1 /vagrant/project/lib/Cake/Console/ShellDispatcher.php(219): Shell->runCommand('test', Array)
#2 /vagrant/project/lib/Cake/Console/ShellDispatcher.php(66): ShellDispatcher->dispatch()
#3 /vagrant/project/app/Console/cake.php(36): ShellDispatcher::run(Array)
#4 {main}

```

File: Lib/SentryErrorHandler.php [hunk 1]
```diff
@@ -1,47 +1,81 @@
...
+                    ));
+                }
+                $eventId = $client->captureException($exception, array(
+                    'extra' => array(
+                        'php_version' => phpversion(),
+                        'class' => get_class($exception)
+                    ),
+                ));
+
+                CakeSession::write('sentry_event_id',$eventId);
```
Comment (mfn) [inline Lib/SentryErrorHandler.php:49]: This is a problem for 2 reasons:
- you might not be in a context where a session is established
- for the same reason, if you don't, you'll be greeted with this error: `Uncaught Error: Class 'CakeSession' not found in /…/app/Plugin/Sentry/Lib/SentryErrorHandler.php:49`

What was the reason to write this to the session here anyway?

This seems highly application specific to me and shouldn't be here in the first place. Or at least, be disabled by default.

**Pull request merged**